packages/client/ui/react-ui/src/providers/signers/CrossmintSignerProvider.tsx

```diff
@@ -258,63 +259,71 @@ export function CrossmintSignerProvider({
                         },
-                        signTransaction: async (transaction: VersionedTransaction) => {
-                            console.log("Signing transaction...", transaction);
-                            const messageData = transaction.message.serialize();
-                            const res = await iframeWindow.current?.sendAction({
-                                event: "request:sign",
-                                responseEvent: "response:sign",
-                                data: {
-                                    authData: {
-                                        jwt,
-                                        apiKey,
-                                    },
+                        signTransaction: (transaction: VersionedTransaction) => {
+                            return signWithRecovery(async () => {
+                                const messageData = transaction.message.serialize();
+                                const res = await iframeWindow.current?.sendAction({
+                                    event: "request:sign",
+                                    responseEvent: "response:sign",
                                     data: {
-                                        keyType: "ed25519",
-                                        bytes: base58.encode(messageData),
-                                        encoding: "base58",
+                                        authData: {
+                                            jwt,
+                                            apiKey,
+                                        },
+                                        data: {
+                                            keyType: "ed25519",
+                                            bytes: base58.encode(messageData),
+                                            encoding: "base58",
+                                        },
                                     },
-                                },
-                                options: DEFAULT_EVENT_OPTIONS,
+                                    options: DEFAULT_EVENT_OPTIONS,
+                                });
+                                if (res?.status === "error") {
+                                    const err = new Error(res.error);
+                                    (err as any).code = res.code;
+                                    throw err;
+                                }
+                                if (res?.signature == null) {
+                                    throw new Error("Failed to sign transaction");
+                                }
+                                transaction.addSignature(
+                                    new PublicKey(publicSignerAddress),
+                                    base58.decode(res.signature)
+                                );
+                                return transaction;
                             });
```

Comment:
Only changes here are:

wrapping each method with `signWithRecovery`, and adding the following error handling for signWithRecovery to catch and handle.
```TS
if (res?.status === "error") {
    const err = new Error(res.error);
    (err as any).code = res.code;
    throw err;
}
```
I consider this approach to be gross, but didn't want to do any big refactors with the context that this section will need to be re-written soon to be unified with the React native SDK.